### PR TITLE
fix: set python container version explicitly

### PR DESCRIPTION
**Pull request by ajberdy**
*Issue #, if available:*

*Description of changes:*

Previous set up was skipping even when version matched

*Testing done:*

integ test passing with correct python version and skipping with incorrect

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/integ_tests/test_create_quantum_job.py [hunk 1]
```diff
@@ -200,7 +201,7 @@ def test_completed_quantum_job(aws_session, capsys):
 
 
 @pytest.mark.xfail(
-    (sys.version_info.major, sys.version_info.minor) != decorator_python_version,
+    (sys.version_info.major, sys.version_info.minor) != (3, 10),
```
Comment (kshitijc) [inline test/integ_tests/test_create_quantum_job.py:204]: I'm not sure I understand why the previous setup doesn't work. Could you please add more details on what is causing that to fail?

File: test/integ_tests/test_create_quantum_job.py [hunk 2]
```diff
@@ -27,12 +27,13 @@
...
-    tag = aws_session.get_full_image_tag(image_uri)
-    major_version, minor_version = re.search(r"-py(\d)(\d+)-", tag).groups()
-    return major_version, minor_version
+# todo: capture python version from container
+# @pytest.fixture
+# def decorator_python_version(aws_session):
+#     image_uri = retrieve_image(Framework.BASE, aws_session.region)
+#     tag = aws_session.get_full_image_tag(image_uri)
+#     major_version, minor_version = re.search(r"-py(\d)(\d+)-", tag).groups()
+#     return major_version, minor_version
```
Comment (ajberdy) [inline test/integ_tests/test_create_quantum_job.py:36]: I'll capture a follow-up to go back to getting the python version from the container in the code, but setting manually to 3.10 to unblock

File: test/integ_tests/test_create_quantum_job.py (hunk 1 above)
Comment (ajberdy) [reply test/integ_tests/test_create_quantum_job.py:204]: Yes, the fixture doesn't get evaluated inside the decorator, so this comparison is something like `(3, 10) != <pytest.fixture-decorator_python_version>`

Not sure how this didn't get caught in its pr, but there was a lot of back and forth with the method for skipping and looks like the final version missed the mark

Comment (kshitijc) [reply test/integ_tests/test_create_quantum_job.py:195]: Does it necessarily need to be a fixture? Can it be instead set to a regular variable?

**Comment by ajberdy** [comment]
Getting an error now with outputs on the integ tests, believe this is also related to the containers change

**Comment by ajberdy** [comment]
I have confirmed that integ tests PASS with 3.10 on the latest commit and have verified skipping with 3.9 since the last change to the xfail decorators

**Comment by ajberdy** [comment]
Reverified passing tests since last commit

File: test/integ_tests/test_create_quantum_job.py [hunk 3]
```diff
@@ -244,6 +244,8 @@ def decorator_job(a, b: int, c=0, d: float = 1.0, **extras):
         with open("test/output_file.txt", "w") as f:
             f.write("hello")
 
+        return job_test_script.job_helper()
```
Comment (ajberdy) [inline test/integ_tests/test_create_quantum_job.py:247]: switched this test to use the return method for more variety

File: test/integ_tests/test_create_quantum_job.py [hunk 4]
```diff
@@ -304,6 +305,7 @@ def decorator_job_submodule():
         ) as f:
             assert f.readlines() == ["pytest\n"]
         assert dir(pytest)
+        save_job_result(submodule_helper())
```
Comment (ajberdy) [inline test/integ_tests/test_create_quantum_job.py:308]: moved this line lower so if something above fails the results assert will fail

**Review by kshitijc — APPROVED** [review]
Do we also need a [test](https://github.com/amazon-braket/amazon-braket-containers/blob/main/test/braket_tests/common/braket_jobs_util.py#L36) for decorators in the containers repo?

**Comment by ajberdy** [comment]
> Do we also need a [test](https://github.com/amazon-braket/amazon-braket-containers/blob/main/test/braket_tests/common/braket_jobs_util.py#L36) for decorators in the containers repo?

Good call, I'll capture a task to add this

**Pull request merged**